forge-game/src/main/java/forge/game/ability/effects/EffectEffect.java

```diff
@@ -211,10 +199,18 @@ public void resolve(SpellAbility sa) {
             }
 
             // Grant Keywords
-            if (effectKeywords != null) {
-                for (final String s : effectKeywords) {
-                    final String actualKeyword = hostCard.getSVar(s);
-                    eff.addIntrinsicKeyword(actualKeyword);
+            if (sa.hasParam("Keywords")) {
+                List<String> effectKeywords = Arrays.asList(sa.getParam("Keywords").split(","));
+                if (sa.hasParam("SharedKeywordsZone")) {
+                    List<ZoneType> zones = ZoneType.listValueOf(sa.getParam("SharedKeywordsZone"));
+                    String[] restrictions = sa.hasParam("SharedRestrictions") ? sa.getParam("SharedRestrictions").split(",")
+                            : new String[]{"Card"};
+                    effectKeywords = CardFactoryUtil.sharedKeywords(effectKeywords, restrictions, zones, hostCard, sa);
+                }
+                if (effectKeywords != null) {
+                    for (final String s : effectKeywords) {
+                        eff.addIntrinsicKeyword(s);
```

Comment:
Having an effect gain keywords feels wrong
Can't you just remember the strings?

Reply:
Sure, could do it that way... the code for putting keywords on Effects was sitting there but unused on any card script, probably just an anachronism

Reply:
I did like the way this version turned out for UI – seemed clean